### ISSUE TITLE
evolution: Let the project be in typescript with proper typings

### ISSUE THREAD

**Issue by tahini** [chairemobilite/evolution#49]
Dependent on chairemobilite/evolution#48 

To reduce the risk of errors in survey development, the projects should be in typescript, with the survey responses properly typed. The widget configurations should also follow a proper API.

**Comment by samuel-duhaime** [comment]
@tahini Can we close? We only have some evolution-legacy in Javascript.

**Comment by tahini** [comment]
> [@tahini](https://github.com/tahini) Can we close? We only have some evolution-legacy in Javascript.

Yes we can close, all our newer projects are in typescript, as well as the demo_survey
